### ISSUE TITLE
Stopping the backend from source leads to frontend still running

### ISSUE THREAD

**Issue by herzenschein** [owncast/owncast#3222]
### Share your bug report, feature request, or comment.

I attempted to build the [backend from source](https://github.com/owncast/owncast#backend) on Linux as a way to test Owncast, and it actually works great. `go run main.go` and the backend works and the frontend shows up fine.

But now, even though I Ctrl+C the process, everytime I go to localhost:8080 it opens the web interface, with an immediate redirect to localhost:8080/#. Which is weird, because Ctrl+Shift+R on localhost:8080/# leads to a "it was not possible to connect" page.

I then tried `go run main.go -webserverport 8181`, stopped the process, and yup, now both localhost:8080 and localhost:8181 open the web interface. So it's easily reproducible.

My guess is that when building the backend from source, there's no call to terminate the frontend whenever the backend process is stopped.

This leads to me being unable to run other docker/podman services on the same port.

**Comment by gabek** [comment]
Just for clarification, when you run the Owncast process on port 8080 there is no separate frontend. If you run a dev version of the web application then it would run standalone, on port 3000. But that's not what you're doing here.

I'm not exactly sure what you're seeing. If I had to guess you're simply seeing cached files as there's a certain amount of pre-caching of assets that happen with the web application to speed component rendering. If the port continues to be used after you kill it, perhaps you're not actually killing it or something else is using it? I'm just throwing out guesses here, however.

So in short owncast is self-contained as one process. Once you kill the Owncast application, it no longer has an active web server to serve frontend assets.

**Comment by herzenschein** [comment]
I had tried deleting `~/.cache`, `/tmp` and even `~/go/.../cache` and tried using Firefox' `Clear Data...` so I had given up about it being a cache issue. It was also weird that there was a redirect, I wasn't aware that was possible with cached data. :eyes: 

After reading your comment I did some more cache digging. Then I found `Manage Data...` in Firefox, found an entry for "localhost" and deleted the entry, and that solved it.

Sorry for the noise and thank you for your time. At least now if someone else faces the same issue they'll find it here! :P
